A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+
+| Principal Field  | gRPC Equivalent |
+| ---------------- | --------------- |
+| authenticated.principal_name | The first URI/DNS SAN; same as Envoy |
+| source_ip        | Peer address for this connection |
+| direct_remote_ip | Peer address for this connection |
+| remote_ip        | Peer address for this connection |
+| header           | Metadata (with caveats) |
+| url_path         | Fully-qualified RPC method name with leading slash |
+| metadata         | Hard-coded as empty; never matches |
+
+TODO: Need to NACK to be consistent on `remote_ip` `source_ip` if
+x-forwarded-for, proxy protocol, etc is configured.
+
+[RBAC policy]: https://github.com/envoyproxy/envoy/blob/10c17a7cd90b013c38dfbfbf715d3c24fdd0477c/api/envoy/config/rbac/v3/rbac.proto
+[RBAC network filter]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/network/rbac/v3/rbac.proto
+
+### Temporary environment variable protection
+
+The environment variable `GRPC_XDS_EXPERIMENTAL_RBAC` will be used to gate
+the feature until it is deemed stable for a particular implementation. If unset
+or not `true`, all RBAC logic presented in this gRFC will not come into effect.
+
+## Rationale
+
+Few design decisions were necessary; RBAC was a pretty strong fit with gRPC's
+needs for an initial authorization policy. The name seems like a misnomer as it
+lacks a level of indirection (the "role") between principals and permissions as
+would be expected from "role based access control", but that is a fair design
+decision and does not impact applicability at this time.
+
+Fuller-fledged authz policy supporting end-user authorization for
+application-specific resources would be useful, but needs to be future work as
+it has multiple technical challenges including need for delayed authz policy
+loading (i.e., delegating to a remote authz server) to be able scale, a way to
+extract the application-level resource from the request, and support for
+authenticating end users.
+
+Envoy also defines an [RBAC network filter][]. Such a filter is helpful for TCP
+or non-HTTP proxying, but could be used for HTTP traffic. Such a network filter
+has the advantage over an HTTP filter as it would 1) have lower ALLOW overhead
+as the check is once per connection and 2) reduce the attack surface to
+malicious clients. When used with HTTP/2, however, avoiding wildly out-of-date
+authz checks would require limiting the lifetime of HTTP/2 connections. In
+addition, there is no way to report clear error messages to the client. Since
+it is unable to support path-based verification it would likely need to be used
+in conjunction with an RBAC HTTP filter for usage in gRPC. It may be useful to
+support one day, but would most likely be a more advanced feature.
+
+## Implementation
+
+[A description of the steps in the implementation, who will do them, and when.  If a particular language is going to get the implementation first, this section should list the proposed order.]
```

Comment:
Any discussion of caches? You can cache results instead of evaluating RBAC for each RPC but we need a design/approach to do that